### DISCUSSION TITLE
Mockk equivalent of Mockito::mockingDetails?

### DISCUSSION THREAD

**Discussion by nkiesel** [mockk/mockk#744] [Q&A]
I'm trying to convert some Kotlin code from Mockito to Mockk.  Everything works nicely, but I am now stuck at my last problem: I need to report how often a specific mocked method was invoked. The Mockito code basically does
```kotlin
private fun invocationCount(methods: Set<Method>): Int {
    return returnedResultSets.flatMap { mockingDetails(it).invocations }.count { it.method in methods }
}
```
`returnedResultSets` is a `List<ResultSet>` which contains all the `ResultSet` mock  objects the code created.

This then gets called with e.g.
```kotlin
setOf(
    ResultSet::class.java.getDeclaredMethod("getDouble", Integer.type)
    ResultSet::class.java.getDeclaredMethod("getDouble", String::class.java)
)
```
Is there any way to implement this for Mockk?

**Comment by Raibaz** [comment]
It shouldn't be too hard: MockK already holds all this information in a CallRecorder object, so implementing a method that exposes it can definitely be done.

**Reply by nkiesel** [reply]
I tried to get to that information using `MockKGateway.implementation` but while I found a call recorder through this, all the counts in there where "0".

Note: we switched from a mockK-mocked implementation to a handcrafted implementation over the weekend (the code in question was really abusing mocking) and thus my immediate need has been resolved. However, we plan to switch most of our current Mockito/PowerMock-based unit tests to MockK, so this problem will resurface again in another context.  Thus, I would still like to find the correct answer.

**Comment by AleksandrRadchenko** [comment]
For anyone as me who is looking how to check if `myBean` is mock, just use `myBean.isMock` in debugger. This triggers extension function `MockkFunctinosKt.isMock`. Otherwise, directly `MockKGateway.implementation().mockFactory.isMock(myBean)`.
